perfmetrics/scripts/ls_metrics/listing_benchmark.py

```diff
@@ -0,0 +1,473 @@
+    format='%(asctime)s [%(levelname)s] %(message)s',
+    handlers=[logging.StreamHandler(sys.stdout)],
+)
+log = logging.getLogger()
+
+
+def OutputResults(folders, results_list, message, num_samples) -> dict:
+  """Outputs the results on the console.
+
+  This function takes in dictionary containing the list of results (for all samples) for each
+  testing folder, for both the gcs bucket and persistent disk. Then it generates various metrics
+  out of these lists and outputs them into the console.
+
+  The metrics present in the output are (in msec):
+  Mean, Median, Standard Dev, 0th %ile, 20th %ile, 40th %ile, 60th %ile, 80th %ile, 90th %ile,
+  95th %ile, 98th %ile, 99th %ile, 100th %ile.
+
+  Args:
+    folders: List containing protobufs of testing folders.
+    results_list: Dictionary containing the list of results (for all samples)
+                  for each testing folder.
+    message: String which describes/titles the test.
+    num_samples: Number of samples to collect for each test.
+
+  Returns:
+    A dictionary containing the various metrics in a JSON format.
+  """
+  metrics = dict()
+
+  for testing_folder in folders:
+    metrics[testing_folder.name] = dict()
+
+    metrics[testing_folder.name]['Test Desc.'] = message
+    metrics[testing_folder.name]['Number of samples'] = num_samples
+    results_list[testing_folder.name] = sorted(results_list[testing_folder.name])
+
+    metrics[testing_folder.name]['Mean'] = stat.mean(results_list[testing_folder.name])
+    metrics[testing_folder.name]['Median'] = stat.median(results_list[testing_folder.name])
+    metrics[testing_folder.name]['Standard Dev'] = stat.stdev(gcs_bucket_results[testing_folder.name])
+
+    metrics[testing_folder.name]['Quantiles'] = dict()
+    for percentile in range(0, 100, 20):
+      metrics[testing_folder.name]['Quantiles']['{} %ile'.format(percentile)] = np.percentile(results_list[testing_folder.name], percentile)
+
+    metrics[testing_folder.name]['Quantiles']['90 %ile'] = np.percentile(results_list[testing_folder.name], 90)
+    metrics[testing_folder.name]['Quantiles']['95 %ile'] = np.percentile(results_list[testing_folder.name], 95)
+    metrics[testing_folder.name]['Quantiles']['98 %ile'] = np.percentile(results_list[testing_folder.name], 98)
+    metrics[testing_folder.name]['Quantiles']['99 %ile'] = np.percentile(results_list[testing_folder.name], 99)
+    metrics[testing_folder.name]['Quantiles']['100 %ile'] = np.percentile(results_list[testing_folder.name], 100)
+
+  print(metrics)
+  return metrics
+
+
+def RecordTimeOfOperation(command, path, num_samples) -> list:
+  """Runs the command on the given path for given num_samples times.
+
+  Args:
+    command: Command to run.
+    path: Path at which to run the command.
+    num_samples: Number of times to run the command.
+
+  Returns:
+    A list containing the latencies of operations in milisecond.
+  """
+  result_list = []
+  for _ in range(num_samples):
+    st = time.time()
```

Comment:
for every time variable its important to append sec or msec in the variable name to know what time we are getting. also dont use shortnames. rename it start_time_sec/end_time_sec

Reply:
done